src/database/nodeinfostorage.cpp

```diff
@@ -0,0 +1,209 @@
+        dstItem.mModelName  = cpuInfo->getValue<std::string>("modelName").c_str();
+        dstItem.mNumCores   = cpuInfo->getValue<size_t>("numCores");
+        dstItem.mNumThreads = cpuInfo->getValue<size_t>("numThreads");
+        dstItem.mArch       = cpuInfo->getValue<std::string>("arch").c_str();
+        dstItem.mArchFamily = cpuInfo->getValue<std::string>("archFamily").c_str();
+
+        dst.PushBack(dstItem);
+    }
+}
+
+Poco::JSON::Array NodeInfoStorage::ConvertPartitionInfoToJSON(const aos::Array<aos::PartitionInfo>& partitionInfo)
+{
+    Poco::JSON::Array dst;
+
+    for (const auto& srcItem : partitionInfo) {
+        Poco::JSON::Object pocoItem;
+        Poco::JSON::Array  types;
+
+        pocoItem.set("name", srcItem.mName.CStr());
+        for (const auto& type : srcItem.mTypes) {
+            types.add(type.CStr());
+        }
+        pocoItem.set("types", types);
+        pocoItem.set("totalSize", srcItem.mTotalSize);
+        pocoItem.set("path", srcItem.mPath.CStr());
+        pocoItem.set("usedSize", srcItem.mUsedSize);
+
+        dst.add(pocoItem);
+    }
+
+    return dst;
+}
+
+void NodeInfoStorage::ConvertPartitionInfoFromJSON(const Poco::JSON::Array& src, aos::Array<aos::PartitionInfo>& dst)
+{
+    for (const auto& srcItem : src) {
+        const auto& partitionInfo = srcItem.extract<Poco::JSON::Object::Ptr>();
+        AssertNotNull(partitionInfo);
+
+        aos::PartitionInfo dstItem;
+
+        dstItem.mName = partitionInfo->getValue<std::string>("name").c_str();
+
+        const auto& types = partitionInfo->get("types").extract<Poco::JSON::Array::Ptr>();
+        AssertNotNull(types);
+        for (const auto& type : *types) {
+            dstItem.mTypes.PushBack(type.convert<std::string>().c_str());
+        }
+
+        dstItem.mTotalSize = partitionInfo->getValue<size_t>("totalSize");
+        dstItem.mPath      = partitionInfo->getValue<std::string>("path").c_str();
+        dstItem.mUsedSize  = partitionInfo->getValue<size_t>("usedSize");
+
+        dst.PushBack(dstItem);
+    }
+}
+
+Poco::JSON::Array NodeInfoStorage::ConvertAttributesToJSON(const aos::Array<aos::NodeAttribute>& attributes)
+{
+    Poco::JSON::Array dst;
+
+    for (const auto& srcItem : attributes) {
+        Poco::JSON::Object pocoItem;
+
+        pocoItem.set("name", srcItem.mName.CStr());
+        pocoItem.set("value", srcItem.mValue.CStr());
+
+        dst.add(pocoItem);
+    }
+
+    return dst;
+}
+
+void NodeInfoStorage::ConvertAttributesFromJSON(const Poco::JSON::Array& src, aos::Array<aos::NodeAttribute>& dst)
+{
+    for (const auto& srcItem : src) {
+        const auto& attribute = srcItem.extract<Poco::JSON::Object::Ptr>();
+        AssertNotNull(attribute);
+
+        aos::NodeAttribute dstItem;
+
+        dstItem.mName  = attribute->getValue<std::string>("name").c_str();
+        dstItem.mValue = attribute->getValue<std::string>("value").c_str();
+
+        dst.PushBack(dstItem);
```

Comment:
Consider adding check for PushBack result

Reply:
done